### PR TITLE
[SofaLoader] ADD normals and vectors to Legacy vtk import

### PR DESCRIPTION
**Pull request by untereiner**
normals and vectors to Legacy vtk import

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp [hunk 1]
```diff
@@ -635,6 +667,34 @@ bool LegacyVTKReader::readFile(const char* filename)
...
+					lnData >> dataName >> dataType;
+					BaseVTKDataIO*  data = newVTKDataIO(dataType, 3);
+					if (data != NULL)
+					{
+						if (data->read(inVTKFile, nb_ele, binary))
+						{
+							inputDataVector.push_back(data);
+							data->name = dataName;
+							if (kw == "CELL_DATA"){
+								msg_info(this) << "Read cell data: " << data->name;
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp]: Small tips: 
writing msg_info() is equivalent to msg_info(this) ...

I also wonder if there is a reason why we print something for "VECTORS" and not for "NORMALS" ?

Comment (untereiner) [reply SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp]: you are right we should do it for both or not at all. What do you prefer ?

**Comment by damienmarchal** [comment]
Hello,

Thank you for the new features. 
@hugtalbot and @guparan would it be possible, before a PR to merged to have it "astyled" to that the 
indentation is consistant ?

File: SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp (hunk 1 above)
Comment (damienmarchal) [reply SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp]: I prefer consistency :)

Comment (untereiner) [reply SofaKernel/modules/SofaLoader/MeshVTKLoader.cpp]: both suggestions are consistent... well I'll do something

**Comment by untereiner** [comment]
@hugtalbot  you should provide a coding style file or reformatting it using clang to avoid annoying contributors with spaces/tabs. Two commits to fix indentation and more than 16 days to merge 20 lines of code is a waste of time

**Comment by damienmarchal** [comment]
My 2 cents...  

Taking "16 days to merge 20 lines of code is a waste of time" is true...and this is the consequence that no-one ever took time to handle that properly. As usual no one's concerns tend to become everyone's waste of time. 

My suggestion to fix that is:
- 0) make a style sheet that matches the Sofa guideline (or change the guideline to match an automated formatter style sheet which is much easier :))
- 1) apply an automated style sheet to format the whole code base. 
- 2) when making PR, everyone should apply the provided style sheet.
- 3) when merging a PR, the automated stylesheet should be applied again. 

But ho wait ! 
Some told me here is already a stylesheet for astyle (to use in qtcreator) since years in the tools/astyle 
directory. 
This may be a good starting point for (0) and (1). 

What about having 0 et 1 done for 17.16 release ?

**Comment by untereiner** [comment]
You are right and it is shitting for the person on whom it falls :)
Haa! A stylesheet! But, as usual with sofa, it takes some time to set it up (install astyle, install the QtCreator plugin, add the configuration file). 
I agree with you to make for 17.16 on the whole repository

**Comment by damienmarchal** [comment]
To anyone (@guparan and @hugtalbot, @fredroy, @bcarrez ) Can you please check how the formatting sheet should be updated to stick the the sofa guidelines (or withdrawn some rule from the guidelines) so we have an up-to date sheet to use.

**Comment by guparan** [comment]
I agree with the style checking + auto-formatting ideas, it has been regularly discussed for a long time now. Here is what I propose for the astyle sheet (based on the existing one):
```bash
# Brace Style
--style=allman

# Indentation
--indent-switches
--indent=spaces=4
--indent-labels
--indent-col1-comments
--max-instatement-indent=100

# Padding
--pad-oper
--pad-comma
--align-pointer=type
--align-reference=type

# Formatting
--break-one-line-headers
--add-braces
--convert-tabs
```
Tell me if this is breaking some of our [coding guidelines](https://github.com/sofa-framework/sofa/blob/master/GUIDELINES.md) but I think it is alright. My only doubt is about forcing or not braces around one line blocks (if, else, while, ...). I personally prefer when they are forced (particularly considering that SOFA devs are changing often and enjoy "quick and dirty" changes a lot) but the Google C++ Style Guide says the opposite [here](https://github.com/sofa-framework/sofa/blob/master/GUIDELINES.md).
What do you guys think?

**Comment by damienmarchal** [comment]
Great... 
Thank guillaume. 

At the present state it is much more important to me that we have consistency either it is with single-line braces or not. So as you have an opinion I vote for yours (as I see the goods of what you are proposing) :) 

Any how you can count on me to use this style sheet. 

EDIT: I will try to make a qtcreator vresion of this one.

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Comment by untereiner** [comment]
[ci-build][with-scene-tests][ci-reformat]

**Comment by untereiner** [comment]
[ci-reformat]

**Comment by untereiner** [comment]
[ci-build][with-scene-tests]

**Comment by damienmarchal** [comment]
Hello @untereiner  
The CI is under heavy crash...so if your build does not complete please wait, with @guparan & @bcarrez we are working at that and have good hopes it will be fixed tomorrow.

**Comment by guparan** [comment]
It should be alright now. Let's try to [ci-build][with-scene-tests] :-)

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests]

**Pull request merged**